### PR TITLE
`hack/check-license-header.sh` ignores folders related to development

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
Follow-up of #7680 
/cc @acumino @oliver-goetz 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: hack/check-license-header.sh [hunk 1]
```diff
@@ -18,7 +18,16 @@ set -e
 
 echo "> Checking if license header is present in all required files."
 
-missing_license_header_files="$(addlicense -check -ignore "vendor/**" -ignore "**/*.md" -ignore "**/*.html" -ignore "**/*.yaml" -ignore "**/Dockerfile" --ignore "hack/tools/gomegacheck/**" --ignore "pkg/operation/botanist/**/*.sh" .)" || true
+missing_license_header_files="$(addlicense -check \
```
Comment (acumino) [inline hack/check-license-header.sh]: Can you adapt the same here https://github.com/gardener/gardener/blob/1e9b2c1a4e5ec47d29b250eaf99a35dad05f525d/hack/add-license-header.sh#L23

**Comment by oliver-goetz** [comment]
How about adding `.vscode/**` to the ignore list too?

**Review by acumino — APPROVED** [review]
/approve

**Pull request merged**